xla/service/algebraic_simplifier.cc

```diff
@@ -1147,6 +1147,15 @@ Status AlgebraicSimplifierVisitor::HandleAnd(HloInstruction* logical_and) {
 }
 
 Status AlgebraicSimplifierVisitor::HandleBitcast(HloInstruction* bitcast) {
+  // It's tricky for the simplifier to determine whether
+  // it should remove the op when control deps are present. I.e.
+  // control deps might be added to preserve a certain order.
+  // It's better to not process in that case.
+  if(bitcast->control_predecessors().size() > 0) {
```

Comment:
looks like space btw `if` and `(` is missing. better to format the code block

Reply:
sorry i just pushed this change to run a test. i will add some tests and formatting before i mark it ready

Reply:
I can format on import.